osf/models/contributor.py

```diff
@@ -41,6 +42,12 @@ class Meta:
         # NOTE: Adds an _order column
         order_with_respect_to = 'node'
 
+    def save(self, *args, **kwargs):
+        if self.is_curator and self.visible:
```

Comment:
I feel like we might have reduced the complexity of this to the point where a DB constraint actually is faster the overriding save.